src/editor/editorui.js

```diff
@@ -60,8 +54,92 @@ export default class EditorUI {
 		 */
 		this.focusTracker = new FocusTracker();
 
+		/**
+		 * Stores all editable elements used by the editor instance.
+		 *
+		 * @protected
+		 * @member {Array.<module:ui/editorui/editoruiview~EditorUIView>}
+		 */
+		this._editableElements = [];
+
+		/**
+		 * **Deprecated** since `v12.0.0`. This property is deprecated and should not be used. Use the property
+		 * from the subclass directly instead, for example
+		 * {@link module:editor-classic/classiceditorui~ClassicEditorUI#_view ClassicEditorUI#_view}.
+		 *
+		 * The main (top–most) view of the editor UI.
+		 *
+		 * @deprecated v12.0.0 This property is deprecated and should not be used. Use the property
+		 * from the subclass directly instead, for example
+		 * {@link module:editor-classic/classiceditorui~ClassicEditorUI#_view ClassicEditorUI#_view}.
+		 * @private
+		 * @member {module:ui/editorui/editoruiview~EditorUIView} #_view
+		 */
+		this._view = view;
+
+		// Check if `view` parameter was passed. It is deprecated and should not be used.
+		if ( view ) {
+			/**
+			 * This error is thrown when  the deprecated `view` parameter is passed to the
+			 * {@link module:core/editor/editorui~EditorUI#constructor EditorUI constructor}. Only subclass (for example
+			 * {@link module:editor-classic/classiceditorui~ClassicEditorUI}) should use it without passing it further.
+			 *
+			 * @error deprecated-editorui-view-param-in-constructor
+			 */
+			log.warn( 'deprecated-editorui-view-param-in-constructor: The EditorUI#constructor `view` parameter is deprecated.' );
+		}
+
 		// Informs UI components that should be refreshed after layout change.
 		this.listenTo( editor.editing.view.document, 'layoutChanged', () => this.update() );
+
+		// Delegate `ready` as `editor.uiReady` event. The `uiReady` is deprecated and should be fired too.
+		this.delegate( 'ready' ).to( this.editor, 'uiReady' );
```

Comment:
I think this should be moved to the `editor` class. It is the editor who handles warnings if someone attaches a listener to this event and it is editors event, so it should take care about it.

Reply:
Since `ui` is a property of `EditorWithUI`, this delegation could not be moved to `Editor` as there is no `ui`. It could be alternatively moved to `Editor` subclasses which uses `EditorWithUI` (basically all editor creators, e.g. `ClassicEditor`) but putting it in every creator doesn't make sense. So we decided to leave it here.